src/settings/settings.go

```diff
@@ -83,11 +83,12 @@ type Settings struct {
 	RuntimeWatchRoot      bool   `envconfig:"RUNTIME_WATCH_ROOT" default:"true"`
 
 	// Settings for all cache types
-	ExpirationJitterMaxSeconds int64   `envconfig:"EXPIRATION_JITTER_MAX_SECONDS" default:"300"`
-	LocalCacheSizeInBytes      int     `envconfig:"LOCAL_CACHE_SIZE_IN_BYTES" default:"0"`
-	NearLimitRatio             float32 `envconfig:"NEAR_LIMIT_RATIO" default:"0.8"`
-	CacheKeyPrefix             string  `envconfig:"CACHE_KEY_PREFIX" default:""`
-	BackendType                string  `envconfig:"BACKEND_TYPE" default:"redis"`
+	ExpirationJitterMaxSeconds        int64   `envconfig:"EXPIRATION_JITTER_MAX_SECONDS" default:"300"`
+	LocalCacheSizeInBytes             int     `envconfig:"LOCAL_CACHE_SIZE_IN_BYTES" default:"0"`
+	NearLimitRatio                    float32 `envconfig:"NEAR_LIMIT_RATIO" default:"0.8"`
+	CacheKeyPrefix                    string  `envconfig:"CACHE_KEY_PREFIX" default:""`
+	BackendType                       string  `envconfig:"BACKEND_TYPE" default:"redis"`
+	IncrementingCacheKeysForOverLimit bool    `envconfig:"INCREMENTING_CACHE_KEYS_FOR_OVER_LIMIT" default:"true"`
```

Comment:
I don't understand what this change does. Can you add documentation? Is this a bug that should just be fixed?

Reply:
Hi @mattklein123,

I have added a new setting called **IncrementingCacheKeysForOverLimit**. The reasons for this is to keep the backward compatibility and in order to fix the bug that I mentioned in the issue https://github.com/envoyproxy/ratelimit/issues/427

I will add the documentation also to get a clear understanding of the behavior if we use IncrementingCacheKeysForOverLimit

Thank you

Reply:
This is what I am fixing from this Pr

Let's say we have grouped routes

1. Group: Foo
   -  /foo/v1/hello-1
2. Group: Bar
   - /bar/v1/hello-1
   - /bar/v1/hello-special

And we need to apply the following rate limits for the group: Bar.

1. Per group -> **10** req/min
2. Special rate limit for the route **/bar/v1/hello-special** in group Bar -> **3** req/min

The expectation is,

1. if we invoke **/bar/v1/hello-special** route more than 3 times it should be rate limited.
2. Then the other route(**/bar/v1/hello-1**) has still 7 req/min quota. 
3. Although we invoke **/bar/v1/hello-special** route 5 times,  that should be rate limited from 3 and for the other 2 invocations it should return 429, and **/bar/v1/hello-1** route still should have 7 req/min quota)

But the current behavior is,

1. If we invoke **/bar/v1/hello-special** route 5 times it is rate limited from the 3 requests and for the next 2 invocations, it returns 429 responses. (Overall 5 invocations but only 3 are valid increments)
2. Then if we invoke **/bar/v1/hello-1** route, it should have the 7 req/min quota. But it has only a 5 req/min quota since we invoke **/bar/v1/hello-special** route 5 times including 2 invalid invocations which return 429 responses.

Reply:
OK this makes sense, though I wonder if this should just be the default. If you want to keep the option please document it in README.md with the other options.

Reply:
Hi @mattklein123,

I document the **INCREMENTING_CACHE_KEYS_FOR_OVER_LIMIT** config from 96185143ba0888f71198ee8f406e572b07d0b86a

Thank you